packages/react/src/components/DetailsList/DetailsRow.base.tsx

```diff
@@ -214,6 +215,9 @@ export class DetailsRowBase extends React.Component<IDetailsRowBaseProps, IDetai
     const isContentUnselectable = selectionMode === SelectionMode.multiple;
     const showCheckbox = selectionMode !== SelectionMode.none && checkboxVisibility !== CheckboxVisibility.hidden;
     const ariaSelected = selectionMode === SelectionMode.none ? undefined : isSelected;
+    const group = getItemGroup(groups, itemIndex, groupNestingDepth);
```

Comment:
I am nervous about this function. Each row should *not*, on its own, take the list of all `groups` and calculate its position in the overall list. `DetailsList` will need to iterate the list of all items and groups *once* upfront, and pass each `DetailsRow` its assigned `rowIndex` explicitly. Otherwise this will be a very nasty performance regression with large, grouped lists.